docs/ansible/coding-conventions.md

```diff
@@ -0,0 +1,669 @@
+
+## 🔴 Spacing addons
+
+# Comparing
+Do not compare to literal True/False.
+✅ ***Good***
+
+```yaml
+when: var
+```
+```yaml
+when: not var
+```
+
+❌ ***Bad***
+```yaml
+when: var == True
+```
+```yaml
+when: var == Yes
+```
+```yaml
+when: var != ""
+```
+
+## Playbook File Extension
+
+All Ansible Yaml files should have a .yml extension (and NOT .YML, .yaml etc).
+
+##Vaults
+
+All Ansible Vault files should have a .vault extension (and NOT .yml, .YML, .yaml etc).
+
+## Use Module synchronize Instead of copy for Large Files
+
+## Use Block-Module
+
+Block can help to organize the code and can enable rollbacks.
+```yaml
+- block:
+    copy:
+      src: critical.conf
+      dest: /etc/critical/crit.conf
+    service:
+      name: critical
+      state: restarted
+  rescue:
+   command: shutdown -h now
+```
+
+## Order in playbook
+
+Playbook definitions should follow this order.
+
+* `name`
+* `hosts`
+* `remote_user`
+* `become`
+* `vars`
+* `pre_tasks`
+* `roles`
+* `tasks`
+
+```yaml
+---
+- name: update root authorized_keys for all machines
+  hosts:
+    - all
+  become: true
+
+  vars:
+    root_keys_users:
+      - user: user1
+        key: |
+          ssh-ed25519 AAAAC3NzaC1lZDI1NTE5AAAAIHkCVF05JvfkrfOOESivOxV4N8+A/EMEkF7/nCQMRoQg
+        enabled: true
+
+  pre_tasks:
+    - name: add custom root users
+      set_fact:
+        root_keys_users: "{{ root_keys_users | union( root_keys_users_custom|default([]) ) }}"
+
+  roles:
+    - role: root-keys
+
+  tasks:
+    - name: print out debug message
+      ansible.builtin.debug:
+        msg: "fee foo faa"
+```
+
+ Why do this?
+
+A common order makes playbooks consistent and easier to read for your dear colleagues. Think of them when you write.
+
+## Order in task declaration
+
+A task should be declared in this order.
+
+* `name`
+* module and the arguments for the module
+* arguments for the task in alphabetic order
+* `tags` at the bottom
+
+```yaml
+---
+- name: unhold packages
+  ansible.builtin.shell: 'apt-mark -s unhold {{ item }} && apt-mark unhold {{ item }}'
+  args:
+    executable: /bin/bash
+  changed_when: apt_mark_unhold.rc == 0 and "already" not in apt_mark_unhold.stdout
+  failed_when: false
+  loop: '{{ packages_ubuntu_unhold }}'
+  register: apt_mark_unhold
+  tags:
+    - apt_unhold
+---
+```
+
+### Why do this?
+
+Reason is the same as "order of playbook". To make tasks more consistent and easier to read. Help your colleagues.
+
+## Air and tabulators
+
+Air, one of the **most important thing** for humans and **for code**! It must be an empty line before `vars`, `pre_tasks`, `roles` and `tasks`, and before each task in the tasks definition. Tabulator stops must be set to two, `2`, spaces.
```

Comment:
2 spaces - on the separate line, add good bad and going first!

then about "air", add good / bad examples